pkg/osquery/runtime/osqueryinstance.go

```diff
@@ -328,41 +338,66 @@ type osqueryFilePaths struct {
 // In return, a structure of paths is returned that can be used to launch an
 // osqueryd instance. An error may be returned if the supplied parameters are
 // unacceptable.
-func calculateOsqueryPaths(rootDir, extensionSocketPath string) (*osqueryFilePaths, error) {
+func calculateOsqueryPaths(opts osqueryOptions) (*osqueryFilePaths, error) {
+
+	// Determine the path to the extension socket
+	extensionSocketPath := opts.extensionSocketPath
+	if extensionSocketPath == "" {
+		extensionSocketPath = socketPath(opts.rootDirectory)
+	}
+
+	extensionAutoloadPath := filepath.Join(opts.rootDirectory, "osquery.autoload")
+
+	osqueryFilePaths := &osqueryFilePaths{
+		pidfilePath:           filepath.Join(opts.rootDirectory, "osquery.pid"),
+		databasePath:          filepath.Join(opts.rootDirectory, "osquery.db"),
+		augeasPath:            filepath.Join(opts.rootDirectory, "augeas-lenses"),
+		extensionSocketPath:   extensionSocketPath,
+		extensionAutoloadPath: extensionAutoloadPath,
+		extensionPaths:        make([]string, len(opts.autoloadedExtensions)),
+	}
+
+	if len(opts.autoloadedExtensions) == 0 {
+		return osqueryFilePaths, nil
+	}
+
 	// Determine the path to the extension
 	exPath, err := os.Executable()
 	if err != nil {
 		return nil, errors.Wrap(err, "finding path of launcher executable")
 	}
 
-	extensionPath := filepath.Join(autoupdate.FindBaseDir(exPath), extensionName)
-	if _, err := os.Stat(extensionPath); err != nil {
-		if os.IsNotExist(err) {
-			return nil, errors.Wrapf(err, "extension path does not exist: %s", extensionPath)
-		} else {
-			return nil, errors.Wrapf(err, "could not stat extension path")
-		}
+	file, err := os.Create(extensionAutoloadPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating autoload file")
 	}
+	defer file.Close()
 
-	// Determine the path to the extension socket
-	if extensionSocketPath == "" {
-		extensionSocketPath = socketPath(rootDir)
+	writer := bufio.NewWriter(file)
```

Comment:
Why use a `bufio.Writer` here? (and not write straight to `file`?)

Reply:
I know it doesn't make any difference at this scale, but bufio should result in fewer sys calls to write to the file. However, I guess since it doesn't make a difference, it's less complex with out bufio, so now writing to the file.